CalibTracker/SiPixelESProducers/plugins/SiPixelTemplateDBObjectESProducer.cc

```diff
@@ -36,48 +35,54 @@ class SiPixelTemplateDBObjectESProducer : public edm::ESProducer  {
 public:
 
   SiPixelTemplateDBObjectESProducer(const edm::ParameterSet& iConfig);
-  ~SiPixelTemplateDBObjectESProducer() override;
   std::shared_ptr<const SiPixelTemplateDBObject> produce(const SiPixelTemplateDBObjectESProducerRcd &);
+
+private:
+  edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> magfieldToken_;
+  edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectRcd> dbTokenDefault_;
+  edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectRcd> dbToken0T_;
+  edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectRcd> dbToken2T_;
+  edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectRcd> dbToken3T_;
+  edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectRcd> dbToken35T_;
+  edm::ESGetToken<SiPixelTemplateDBObject, SiPixelTemplateDBObjectRcd> dbToken4T_;
 };
 
 
 SiPixelTemplateDBObjectESProducer::SiPixelTemplateDBObjectESProducer(const edm::ParameterSet& iConfig) {
-	setWhatProduced(this);
+  setWhatProduced(this)
+    .setConsumes(magfieldToken_)
+    .setConsumes(dbTokenDefault_, edm::ESInputTag{"", ""})
+    .setConsumes(dbToken0T_, edm::ESInputTag{"", "0T"})
+    .setConsumes(dbToken2T_, edm::ESInputTag{"", "2T"})
+    .setConsumes(dbToken3T_, edm::ESInputTag{"", "3T"})
+    .setConsumes(dbToken35T_, edm::ESInputTag{"", "35T"})
+    .setConsumes(dbToken4T_, edm::ESInputTag{"", "4T"});
 }
 
 
-SiPixelTemplateDBObjectESProducer::~SiPixelTemplateDBObjectESProducer(){
-}
-
-
-
-
 std::shared_ptr<const SiPixelTemplateDBObject> SiPixelTemplateDBObjectESProducer::produce(const SiPixelTemplateDBObjectESProducerRcd & iRecord) {
-	
-	ESHandle<MagneticField> magfield;
-	iRecord.getRecord<IdealMagneticFieldRecord>().get(magfield);
+	const auto& magfield = iRecord.get(magfieldToken_);
 
 	GlobalPoint center(0.0, 0.0, 0.0);
-	float theMagField = magfield.product()->inTesla(center).mag();
+	float theMagField = magfield.inTesla(center).mag();
```

Comment:
please use ```nominalValue()``` as in ```https://cmssdt.cern.ch/lxr/ident?_i=nominalValue```

Reply:
I think that change would be better done by DPG in a separate PR (this one will likely stay open for a while).